### PR TITLE
feat(orchestrator): add tick-based keepalive and dead peer detection

### PR DESCRIPTION
**Pull request by jtaylor-orbis**
## Summary

Implements two-layer failure detection for peers: immediate `onRpcBroken` (primary) + tick-based keepalive with hold timer expiry (backup for silent failures like network partitions).

**Problem:** The orchestrator had no keepalive, no hold timer, and no dead peer detection. If a peer silently fails, its routes remain in the table indefinitely. The protocol docs (`BGP_PROTOCOL.md`, `INTERNAL_PEERING.md`) describe hold time (180s) and keepalive (every 60s), but none of it was implemented.

**Solution:** A single `setInterval` fires `system:tick` through the normal dispatch pipeline. The RIB handles Tick like any other action in `plan()`:

1. For each established peer: if `now - lastReceived > holdTime` → expire peer, withdraw all its routes
2. For each established peer: if `now - lastSent > holdTime / 3` → generate keepalive propagation
3. Process expirations before keepalives (don't keepalive a dead peer)

**Injectable `now`:** The Tick action carries `now` as a parameter instead of calling `Date.now()` inside the RIB. This makes all keepalive/expiry logic deterministic and testable without mocking timers.

**Changes:**
- `packages/routing/src/state.ts` → added `lastReceived`, `lastSent`, `holdTime` (all optional) to `PeerRecordSchema`
- `apps/orchestrator/src/rib.ts` → handles Tick action: hold timer checks, peer expiry with route withdrawal, keepalive generation (+117 lines). Updates `lastReceived` on Open/Connected/Update, `lastSent` on commit
- `apps/orchestrator/src/orchestrator.ts` → `startTick()`/`stopTick()` with auto-derived interval (`min(holdTime) / 6`)
- `apps/orchestrator/src/peer-transport.ts` → added `sendKeepalive()` method
- `apps/orchestrator/tests/keepalive.test.ts` → 13 tests with injectable `now`

## Context

Part 7 of 8 in the orchestrator refactor stack. See `docs/architecture-proposal-orchestrator-refactor.md` § "Peer Lifecycle and Keepalive" for the 4-state FSM and two-layer detection design.

## Test plan

- [x] New: 13 keepalive tests (peer expiry, keepalive generation, route withdrawal on expiry, timestamp tracking, multiple peer scenarios, hold timer edge cases)
- [x] All existing tests pass (Tick is no-op when no peers have holdTime set — backward compatible)
- [x] All container integration tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by jtaylor-orbis** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/359?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#360** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#359** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/359?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/359?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#358** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#357** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/357?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#356** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/356?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#355** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/355?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#354** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/354?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#353** <a href="https://app.graphite.com/github/pr/orbisoperations/catalyst-router/353?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/orchestrator/src/rib.ts [hunk 1]
```diff
@@ -549,6 +607,39 @@ export class RoutingInformationBase {
...
+        const propagations: Propagation[] = []
+
+        // 1. Withdrawals for expired peers (processed first)
+        const expiredPeerNames = prevState.internal.peers
+          .filter(
+            (p) =>
+              p.connectionStatus === 'connected' &&
+              p.holdTime != null &&
+              p.lastReceived != null &&
+              now - p.lastReceived > p.holdTime * 1000
```
[the remark this answers is not in the document]
Comment (jtaylor-orbis) [reply apps/orchestrator/src/rib.ts]: Fixed in PR #433: [fix(orchestrator): add holdTime > 0 guards for RFC 4271 compliance](https://github.com/orbisoperations/catalyst-router/pull/433)

Added `holdTime > 0` guard at all three timer check sites (computeNewState expiry, computePropagations expiry filter, keepalive scheduling). Tests updated to assert correct RFC 4271 §4.2 semantics.
